authenticate.js

```diff
@@ -1,23 +1,38 @@
 const alfy = require('alfy');
-const bitbucket = require('./bitbucket/core').bitbucket;
+const {bitbucket} = require('./bitbucket/core');
 
 const ACCESS_TOKEN = 'access_token';
 const GRANT_TYPE = 'client_credentials';
 const ALIVE_TIME = 216000;
 const URL = 'https://bitbucket.org/site/oauth2/access_token';
 
+/**
+ * Generates a HTTP Basic Auth token for accessing the BitBucket API via App Passwords
+ *
+ * @param username {string}
+ * @param password {string}
+ * @returns {string}
+ */
 function getAppPasswordToken(username, password) {
     return `Basic ${Buffer.from([username, password].join(':')).toString('base64')}`;
 }
 
-async function getAuthToken(clientId, secret) {
+/**
+ * Generates a Bearer token for accessing the BitBucket API via OAuth2 Applications
+ *
+ * @param clientId
+ * @param secret
+ * @returns {Promise<string>}
+ */
+async function getOAuthToken(clientId, secret) {
     const accessToken = alfy.cache.get(ACCESS_TOKEN);
 
     if (accessToken) {
-        return bitbucket(accessToken);
+        return accessToken;
     }
 
     try {
+        // Request an access token from BitBucket using the OAuth2 flow
```

Comment:
I'm not a fan of middle code comment, just remove this one as the function's documentation you added (thank you for that 👍 ) says the same

Reply:
Sure thing. I did that just to call attention to the line that actually makes the API call. Probably could have worded it better (I'm bad at documentation lol)